### PR TITLE
Resolve #122

### LINKED ISSUE #122: AjaxFileUpload not working inside asp:WizardStep control
AjaxFileUpload not working inside asp:WizardStep control. I have read about this issue. The workaround by adding a `<div>` container.

Can this be fixed in future releases?

### PR DESCRIPTION
**Pull request by MikhailTymchukDX**
fix upload for hidden AjaxFileUpload

### CODE REVIEW

**Comment by AlekseyMartynov** [comment]
I see that you moved processing from OnPreRender to the earlier stage, OnLoad.
OnInit would be even better. The earlier we interrupt WebForms page lifecycle the better.

File: AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs [hunk 1]
```diff
@@ -231,19 +231,22 @@ public bool AutoStartUpload
...
                     IsInFileUploadPostBack = true;
+
+                ProcessRequest();
             }
         }
 
         protected override void OnLoad(EventArgs e) {
             base.OnLoad(e);
-
+
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs]: "Pull request must not change the formatting of the existing code not related to the pull request." :smile:

**Comment by MikhailTymchukDX** [comment]
Should be OK now.

**Comment by AlekseyMartynov** [comment]
One more precaution. During OnInit stage we should not rely on the control's properties (because they may not be loaded from ViewState yet). How about making `ProcessRequest` and `ParseRequest` static with `Page` passed as an argument?

**Comment by MikhailTymchukDX** [comment]
We can implement it for `ParseRequest()` (narrowing the parameter down to `Page.Request` instead of `Page`).
Making `ProcessRequest()` static will force us to make all `XhrXXX()` methods static, but some of the methods use non-static class members, so we cannot do this.

**Comment by AlekseyMartynov** [comment]
I tried to isolate upload request processing to a class.
https://github.com/AlekseyMartynov/AjaxControlToolkit/commit/d5591dd40110c21f16d8cef48c07b05c57b4ec49

What do you think?
P.S. untested, just made it compile.

**Comment by MikhailTymchukDX** [comment]
I used a slightly different approach when extracting a class, but this style looks better. I will test it.

**Comment by AlekseyMartynov** [comment]
I suggest one more improvement. Let's remove code duplication in lines 235-239 and 532-534. Perform the check in advance and don't call `UploadRequestProcessor` at all if not `IsInFileUploadPostBack`. And there won't be need to pass `ClientID`.

**Comment by MikhailTymchukDX** [comment]
Done.

File: AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs [hunk 2]
```diff
@@ -231,14 +231,32 @@ public bool AutoStartUpload
...
+                UploadStart = UploadStart,
+                UploadComplete = UploadComplete,
+                UploadCompleteAll = UploadCompleteAll,
+                SetUploadedFilePath = SetUploadedFilePath
+            };
+
+            processor.ProcessRequest();
+        }
+
+        bool IsControlRequest() {
```
Comment (AlekseyMartynov) [inline AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs]: `IsControlRequest` - need better name. It should be connected to `IsInFileUploadPostBack`.
Maybe `HasFileUploadPostBackParams`

**Comment by AlekseyMartynov** [comment]
Looks good.

**Pull request merged**